Quaver.Shared/Database/Maps/MapDatabaseCache.cs

```diff
@@ -199,8 +199,22 @@ public static void UpdateMap(Map map)
         /// </summary>
         private static IEnumerable<Map> LoadOsuBeatmapDatabase()
         {
-            var db = OsuDb.Read(ConfigManager.OsuDbPath.Value);
-            MapManager.OsuSongsFolder = Path.GetDirectoryName(ConfigManager.OsuDbPath.Value) + "/Songs/";
+            var dbpath = ConfigManager.OsuDbPath.Value;
+
+            Logger.Debug($"Osu database file {dbpath}", LogType.Runtime);
+
+            if (!File.Exists(dbpath))
+            {
+                Logger.Warning("Osu db path does not exist, returning empty list", LogType.Runtime);
+                return (new List<Map>());
+            }
+
+            var db = OsuDb.Read(dbpath);
+
+            // you need the trailing \ or something in map manager shits the bed and you can't play
+            MapManager.OsuSongsFolder = ConfigManager.OsuSongsDirectory.Value.EndsWith("\\") ? ConfigManager.OsuSongsDirectory.Value : ConfigManager.OsuSongsDirectory.Value + "\\";
```

Comment:
Little suggestion

This might interfere with cross-platform support unless there is a better approach to this, but.. osu! does store values in the registry editor, it might be a good idea for now to find the **osu!.db** file using the **UninstallString** field and then just cutting off the `osu!.exe -uninstall` part

![image](https://user-images.githubusercontent.com/6015313/50377197-eaa14300-0653-11e9-8359-6ca74abfb15d.png)

Reply:
This seems like a decent way to auto-detect the installation. We can try to auto-set the binded config value if it doesn't exist? Saves the trouble of having to edit config for most users since they probably haven't changed the default installation path. 

A simple option to specify the path in-game would be fine too, but I want to refrain from having an option for this due to us wanting to be respectful about not mentioning any of osu!'s trademarks. That's why this seems more-or-less like a hidden option at its current state.

Reply:
Fair enough, if you are wanting to keep away from osu!'s trademark, it would be best in any case to just not have osu! beatmap conversion in place, either using the config, or a proposed alternative.

If anything, an external conversion program could be made to substitute this problem.

Reply:
It should probably be removed when we have an available editor, but I guess its fine for now. Quaver.API provides enough to make an external tool that does the same thing.

Reply:
Another idea would be to contact the osu! legality team and ask for some form of permission for this feature specifically (i assume just trademark).

Reply:
To access the registry we need an extra dependency, see [Accessing registry from managed app?](https://social.technet.microsoft.com/Forums/en-US/d0f15e6d-e83a-4d04-b21d-ed1efe509c56/accessing-registry-from-managed-app)

But I am worried about this breaking cross platform compatibility

Accessing the registry on other platforms can be avoided in the actual code like so
`            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
            {
                // windows specific osu detection
            }
`
But would it fail to build on another platform because import doesn't exist?

Reply:
After second thought, I think manually setting the db path is the good way to go. I've already fixed the crash when the path couldn't be found. I don't think it's a good idea to be messing around with the registry - especially if it's coming to other games/something we have no control over.